### ISSUE TITLE
Set Temporal Namespace

### ISSUE THREAD

**Issue by aakarim** [airbytehq/airbyte#18853]
## Tell us about the problem you're trying to solve
I  have an existing Temporal cluster that I use by setting `TEMPORAL_HOST` in the Airbyte Kustomize definitions. I can use my existing Temporal cluster, but I can't seem to set the namespace, it always runs in the default namespace. This conflicts with my other projects and makes it difficult to see what Airbyte is up to.

I can imagine this is also relevant for users of Temporal Cloud, although I'm not one of them.

## Describe the solution you’d like
An environment variable for all components that access Temporal to set a namespace.

## Describe the alternative you’ve considered or used
I've looked at the source and it seems like the namespace is hardcoded, so overriding would require a fork.

## Additional context
None.

## Are you willing to submit a PR?
I wouldn't know where to start!

**Comment by cpdeethree** [comment]
We no longer support Kustomize. Please let us know if this is an issue still in Helm

**Comment by kenske** [comment]
This still applies to Helm. I don't see an option to specify the temporal namespace in Helm.

**Comment by octavia-squidington-iii** [comment]
At Airbyte, we seek to be clear about the project priorities and roadmap.  This issue has not had any activity for 180 days, suggesting that it's not as critical as others.  It's possible it has already been fixed. It is being marked as stale and will be closed in 20 days if there is no activity. To keep it open, please comment to let us know why it is important to you and if it is still reproducible on recent versions of Airbyte.

**Comment by kenske** [comment]
This is still an issue. It's important to have this feature so that we don't have to duplicate infrastructure for temporal.

**Comment by sepsi77** [comment]
Yes, this is important. I would prefer not to have two Temporal clusters running.

**Comment by talha-naeem1** [comment]
Facing the same problem, we should expose some environment variable to choose different temporal namespace.

**Comment by dkravetz** [comment]
Has there been any progress on this topic? I am currently trying to configure Airbyte to use an existing cluster in order to save on resources, but clashing on (Temporal's) namespace usage doesn't seem like the best idea

**Comment by lerminou** [comment]
Hi, same here, I'm already using a temporal instance

**Comment by lerminou** [comment]
In the meantime, I have a workaound to set temporal_cloud_* variable and point to my local temporal.

because in the temporal_cloud mode, we can choose the namespace.

Shame.

**Comment by jalaziz** [comment]
The issue with using temporal cloud mode is you lose the ability to control retention via the Airbyte config.

Ideally, Airbyte would support setting the namespace for self-hosted Airbyte deployments so that we can use a shared Temporal cluster AND allow Airbyte to control its own reteion.

**Comment by jalaziz** [comment]
I would love to contribute a fix for this, it looks simple enough, but Airbyte no longer accepts contributions to the platform 😦

**Comment by kenske** [comment]
@jalaziz where did you see that?

edit: found it: https://docs.airbyte.com/platform/contributing-to-airbyte#contributions-we-dont-accept

Is airbyte undergoing enshittification now?

**Comment by diseku** [comment]
any progress on it? it's overriding the retention for the `default` namespace via `TEMPORAL_HISTORY_RETENTION_IN_DAYS`

no-go if using a shared cluster
